traffic_ops_ort/traffic_ops_t3c/util/util.go

```diff
@@ -0,0 +1,383 @@
+//
+// if the return code is 1, the the 'name' queried for is not part of a
+//   package or is not installed.
+//
+// otherwise, if the return code is not 0 or 1 and error is set, a general
+// rpm command execution error is assumed and the error is returned.
+func PackageInfo(cmdstr string, name string) ([]string, error) {
+	var result []string
+	switch cmdstr {
+	case "cfg-files": // returns a list of the package configuration files.
+		output, rc, err := ExecCommand("/bin/rpm", "-q", "-c", name)
+		if rc == 1 { // rpm package for 'name' was not found.
+			return nil, nil
+		} else if rc == 0 { // add the package name the file belongs to.
+			log.Debugf("output from cfg-files query: %s\n", string(output))
+			files := strings.Split(string(output), "\n")
+			for ii := range files {
+				result = append(result, strings.TrimSpace(files[ii]))
+			}
+			log.Debugf("result length: %d, result: %s\n", len(result), string(output))
+		} else if err != nil {
+			return nil, err
+		}
+	case "file-query": // returns the rpm name that owns the file 'name'
+		output, rc, err := ExecCommand("/bin/rpm", "-q", "-f", name)
+		if rc == 1 { // file is not part of any package.
+			return nil, nil
+		} else if rc == 0 { // add the package name the file belongs to.
+			log.Debugf("output from file-query: %s\n", string(output))
+			result = append(result, string(strings.TrimSpace(string(output))))
+			log.Debugf("result length: %d, result: %s\n", len(result), string(output))
+		} else if err != nil {
+			return nil, err
+		}
+	case "pkg-provides": // returns the package name that provides 'name'
+		output, rc, err := ExecCommand("/bin/rpm", "-q", "--whatprovides", name)
+		log.Debugf("pkg-provides - name: %s, output: %s\n", name, output)
+		if rc == 1 { // no package provides 'name'
+			return nil, nil
+		} else if rc == 0 {
+			pkgs := strings.Split(string(output), "\n")
+			for ii := range pkgs {
+				result = append(result, strings.TrimSpace(pkgs[ii]))
+			}
+		} else if err != nil {
+			return nil, err
+		}
+	case "pkg-query": // returns the package name for 'name'.
+		output, rc, err := ExecCommand("/bin/rpm", "-q", name)
+		if rc == 1 { // the package is not installed.
+			return nil, nil
+		} else if rc == 0 { // add the rpm name
+			result = append(result, string(strings.TrimSpace(string(output))))
+		} else if err != nil {
+			return nil, err
+		}
+	case "pkg-requires": // returns a list of packages that requires package 'name'
+		output, rc, err := ExecCommand("/bin/rpm", "-q", "--whatrequires", name)
+		if rc == 1 { // no package reuires package 'name'
+			return nil, nil
+		} else if rc == 0 {
+			pkgs := strings.Split(string(output), "\n")
+			for ii := range pkgs {
+				result = append(result, strings.TrimSpace(pkgs[ii]))
+			}
+		} else if err != nil {
+			return nil, err
+		}
+	}
+	return result, nil
+}
+
+func RandomDuration(max time.Duration) time.Duration {
+	rand.Seed(time.Now().UnixNano())
+	return time.Duration(rand.Int63n(int64(max)))
```

Comment:
This panics if `--dispersion=0` is set.
Should this just return 0 if `max` is 0?

Reply:
I've fixed this.  Had a math issue